### ISSUE TITLE
The format of data files is not the plain text

### ISSUE THREAD

**Issue by martinsluka** [daem-on/wtherion#25]
The format of data files is not the plain text as in Therion. It has many reasons in Therion.

**Comment by daem-on** [comment]
Hello, thanks for creating this issue.

All files that you can produce with wtherion **are text files**. Depending on what you mean by "plain text", they are not plain text files, meaning they have a set format that can be parsed by machines. Therion's `.th` and `.th2` files **also have a set format**, so by that definition they are not "plain text".

There are two ways you can produce files in wtherion:
- saves, which are text files in the JSON format, and
- exports, which are text files in the .th2 format

The .th2 format is certainly more human-readable than JSON, but you can edit both of them with a simple text editor (like XTherion's text editor interface).

Here's which buttons correspond to saves and exports in wtherion:

![image](https://github.com/daem-on/wtherion/assets/6278171/106ded9f-22ab-4702-a83f-873e3fc9d7fa)

If you want a .th2 file you should click *Export*. **This is basically the same as saving in XTherion**.

Here's an example of your biely_dom_3 file as a save, and as an export:

[biely_dom_3.zip](https://github.com/daem-on/wtherion/files/15143127/biely_dom_3.zip)

The .json file is the save, the .th2 file is an export.

Hope this clears things up.

**Comment by martinsluka** [comment]
There is the problem. Exported .th2 is different from original .th2. Check attached screenshot.

Martin

￼

**Comment by daem-on** [comment]
Yes, if you import .th2 files and export them, it's not guaranteed that you will get the same file, although they will be in the same format and they will be text files. It's also guaranteed that they will have the same content (semantically).

I can't see the screenshot you've attached, I think because you sent it as an e-mail, please try uploading it on GitHub. If the screenshot shows an actual difference in content and not just formatting, that's an error in wtherion that can be fixed.

I've also implemented an option in wtherion to have the exported .th2 files be closer to XTherion output (indentation, number formatting), which you can use if you also work with XTherion and want minimal changes between them. You can enable this option in *File > Configuration > Export Format*.

Hope this helps.

**Comment by martinsluka** [comment]
This could be a problem in large projects, when several people work on it and share data by VCS as Git. But thank you for improved export of  .th2 files. MartinOdesláno z iPhonu29. 4. 2024 v 9:28, Zih Csongor ***@***.***>:﻿
Yes, if you import .th2 files and export them, it's not guaranteed that you will get the same file, although they will be in the same format and they will be text files. It's also guaranteed that they will have the same content (semantically).
I can't see the screenshot you've attached, I think because you sent it as an e-mail, please try uploading it on GitHub. If the screenshot shows an actual difference in content and not just formatting, that's an error in wtherion that can be fixed.
I've also implemented an option in wtherion to have the exported .th2 files be closer to XTherion output (indentation, number formatting), which you can use if you also work with XTherion and want minimal changes between them. You can enable this option in File > Configuration > Export Format.
Hope this helps.
